source/MaterialXRenderGlsl/GlslRenderer.cpp

```diff
@@ -202,6 +202,7 @@ void GlslRenderer::render()
                         MeshIndexBuffer& indexData = part->getIndices();
                         glDrawElements(GL_TRIANGLES, (GLsizei)indexData.size(), GL_UNSIGNED_INT, (void*)0);
                     }
+                    _program->unbindGeometry();
```

Comment:
I don't think you need this inside the mesh `for` loop as `bindMesh()` should unbind the previous geometry. Suggest to move outside the loop to unbind the last geometry.

Reply:
@kwokcb I went back and checked `bindMesh()` I don't see it unbinding the previous geometry
https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXRenderGlsl/GlslProgram.cpp#L325
Am I missing something?

Reply:
However, it looks like the GlslMaterial has that logic:
https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXRenderGlsl/GlslMaterial.cpp#L108-L121

Reply:
@kwokcb We can update GlslProgram::bindMesh to unbindGeometry(), I don't have any preference.

Reply:
Your right. I misread where the logic was. Sounds like a state variable for the currently bound mesh can to be added here and to unbind when the new mesh differs from the currently bound one. 

Seems to be the "safest" route without changing `GlslProgram` behaviour to avoid any possible side-effects.

How does that sound?

Reply:
Your right. I misread where the logic was. Sounds like a state variable for the currently bound mesh can to be added here and to unbind when the new mesh differs from the currently bound one. 

Seems to be the "safest" route without changing `GlslProgram` behaviour to avoid any possible side-effects.

How does that sound?

Reply:
@kwokcb PTAL at https://github.com/AcademySoftwareFoundation/MaterialX/pull/1299